snuba/query/processors/object_id_rate_limiter.py

```diff
@@ -0,0 +1,96 @@
+from snuba.clickhouse.query_dsl.accessors import get_object_ids_in_query_ast
+from snuba.query.logical import Query
+from snuba.query.processors import QueryProcessor
+from snuba.request.request_settings import RequestSettings
+from snuba.state import get_configs
+from snuba.state.rate_limit import (
+    ORGANIZATION_RATE_LIMIT_NAME,
+    PROJECT_RATE_LIMIT_NAME,
+    RateLimitParameters,
+)
+
+
+class ObjectIDRateLimiterProcessor(QueryProcessor):
+    """
+    A generic rate limiter that searches a query for conditions on the given column.
+    The values in that column are assumed to be an integer ID.
+    """
+
+    def __init__(
+        self,
+        object_column: str,
+        rate_limit_name: str,
+        per_second_name: str,
+        concurrent_name: str,
+    ) -> None:
+        self.object_column = object_column
+        self.rate_limit_name = rate_limit_name
+        self.per_second_name = per_second_name
+        self.concurrent_name = concurrent_name
+
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+        # If the settings don't already have an object rate limit, add one
+        existing = request_settings.get_rate_limit_params()
+        for ex in existing:
+            if ex.rate_limit_name == self.rate_limit_name:
+                return
+
+        obj_ids = get_object_ids_in_query_ast(query, self.object_column)
+        if not obj_ids:
+            return
+
+        # TODO: Add logic for multiple IDs
+        obj_id = obj_ids.pop()
+
+        orl, ocl = get_configs(
```

Comment:
nit: I would prefer to just have the full name of these variables. 
```suggestion
        object_per_second_limit, object_concurrent_limit = get_configs(
```